pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol

```diff
@@ -142,33 +142,22 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, ReentrancyG
             _tokenState[token] = ManagedPoolTokenLib.setTokenScalingFactor(bytes32(0), token);
         }
 
-        // This bytes32 holds a lot of the core Pool state which is read on most interactions, by keeping it in a single
-        // word we can save gas from unnecessary storage reads. It includes items like:
-        // - Swap fees
-        // - Weight change progress
-        // - Various feature flags
-        bytes32 poolState;
-
-        poolState = _startGradualWeightChange(
-            poolState,
+        // We write the pool state here, as both `_setSwapEnabled` and `_setMustAllowlistLPs` read it from storage.
```

Comment:
```suggestion
```

I think we can get rid of this. We only had this comment because we were moving from manipulating the state in memory to reading from storage.